R/parse-body.R

```diff
@@ -483,23 +485,39 @@ parser_rds <- function(...) {
 #' @describeIn parsers feather parser. See [arrow::read_feather()] for more details.
 #' @export
 parser_feather <- function(...) {
-  parser_read_file(function(tmpfile) {
-    if (!requireNamespace("arrow", quietly = TRUE)) {
-      stop("`arrow` must be installed for `parser_feather` to work")
+  parser_read_file(
+    function(tmpfile) {
+      if (!requireNamespace("arrow", quietly = TRUE)) {
+        stop("`arrow` must be installed for `parser_feather` to work")
+      }
+      arrow::read_feather(tmpfile, ...)
+    },
+    function(value, tmpfile) {
+      if (!requireNamespace("readr", quietly = TRUE)) {
+        stop("`readr` must be installed for `parser_feather` to work")
+      }
+      readr::write_file(value, tmpfile)
     }
-    arrow::read_feather(tmpfile, ...)
-  })
+  )
 }
 
 #' @describeIn parsers parquet parser. See [arrow::read_parquet()] for more details.
 #' @export
 parser_parquet <- function(...) {
-  parser_read_file(function(tmpfile) {
-    if (!requireNamespace("arrow", quietly = TRUE)) {
-      stop("`arrow` must be installed for `parser_parquet` to work")
+  parser_read_file(
+    function(tmpfile) {
+      if (!requireNamespace("arrow", quietly = TRUE)) {
+        stop("`arrow` must be installed for `parser_parquet` to work")
+      }
+      arrow::read_parquet(tmpfile, ...)
+    },
+    function(value, tmpfile) {
+      if (!requireNamespace("readr", quietly = TRUE)) {
+        stop("`readr` must be installed for `parser_parquet` to work")
+      }
+      readr::write_file(value, tmpfile)
```

Comment:
What advantage is there to use `readr::write_file()` over `writeBin()`?

Reply:
Do you have an example of a _rare_ issue?

Reply:
I was trying to write a test to for the issue (I thought) I was having with the `writeBin` implementation. The original data I was having this issue (#854) with is not public, and some alternatives I've tried didn't seem to reproduce either. 

I've just gone back to the machine I was trying this on, and it doesn't reproduce there either -- I think this must've been caused by something else. I'll close this PR for now.